pkg/shared/util/version_mapping.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2022 The Numaproj Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"github.com/numaproj/numaflow-go/pkg/info"
+)
+
+type sdkConstraints map[info.Language]string
+
+var versionMappingConfig = map[string]sdkConstraints{
+	"1.2.0-rc1": {
+		info.Go:     "0.7.0-0",
+		info.Python: "0.7.0a",
+		info.Java:   "0.7.1-0",
+	},
+	"1.2.0-rc2": {
+		info.Go:     "0.7.0-0",
+		info.Python: "0.7.0a",
+		info.Java:   "0.7.1-0",
+	},
+	"1.2.0-rc3": {
+		info.Go:     "0.7.0-0",
+		info.Python: "0.7.0a",
+		info.Java:   "0.7.1-0",
+	},
+}
```

Comment:
should we just maintain a YAML file for this and embed that into the build?

Reply:
if we maintain it in a yaml file then we still need to read it into a map so it seems redundant. Also we would need to make the file somehow available to the pods because we cant just do os.ReadFile. Wdyt Vigith?

Reply:
use go:embed ?

i am okay with code too

Reply:
we will use the code for now.